core/src/builder/sighash.rs

```diff
@@ -78,12 +77,23 @@ pub struct PartialSignatureInfo {
     pub kickoff_utxo_idx: usize,
 }
 
+/// Contains information about the spend path of the utxo.
+/// If it is KeyPath, it also includes the merkle root hash of the scripts as
+/// the hash is needed to tweak the key before signing.
+#[derive(Copy, Clone, Debug)]
+pub enum TapTweakData {
```

Comment:
I thought the purpose of this struct is to provide the information required to sign the UTXO (i.e. information required for a tweak or if a tweak is not required). It doesn't provide the spend path. Can we elaborate this in the docs?